stable/external-dns/templates/_helpers.tpl

```diff
@@ -120,6 +120,8 @@ Return true if a secret object should be created
     {{- true -}}
 {{- else if and (eq .Values.provider "cloudflare") (or .Values.cloudflare.apiToken .Values.cloudflare.apiKey) (not .Values.cloudflare.secretName) -}}
     {{- true -}}
+{{- else if and (eq .Values.provider "designate") (or .Values.designate.username .Values.designate.password) -}}
```

Comment:
this line suggests that either the username and/or password HAS to be provided. The README changes indicate both username and password are optional. If it were optional then,

```suggestion
{{- else if (eq .Values.provider "designate") -}}
```

Reply:
The chart doesn’t currently create any secrets for the designate provider.

Reply:
To expand on the comment above, since the chart doesn't currently create any secrets for designate, secrets should only be created if either designate.username or designate.password are specified.

Reply:
I misunderstood this change. I did not see the context of this change